### PR TITLE
Bump `@metamask/inpage-providers` to `@metamask/providers`

### PR DESCRIPTION
**Pull request by adonesky1**
Successfully building, node version at 16. 

Node v18 introduces new errors that will likely require webpack version bump to v5 which is trickier

### CODE REVIEW

File: src/inpage/webpack.config.js [hunk 1]
```diff
@@ -12,15 +12,15 @@ const config = {
   module: {
     rules: [
       {
-        test: /\.(js|jsx)$/u,
-        exclude: /node_modules/u,
```
Comment (jiexi) [inline src/inpage/webpack.config.js:16]: oof. this is unideal. I guess we can kick this can down the road though

**Review by jiexi — DISMISSED** [review]
LGTM. i'll look into the failed CI pipeline

File: src/inpage/webpack.config.js (hunk 1 above)
Comment (adonesky1) [reply src/inpage/webpack.config.js:16]: also works if we do:
```
include: [
          path.resolve(__dirname, '..'),
          path.resolve(__dirname, '..', '..', 'node_modules', '@metamask'),
          path.resolve(__dirname, '..', '..', 'node_modules', 'superstruct'),
        ],
```
but...

File: package.json [hunk 2]
```diff
@@ -57,5 +58,8 @@
     "readable-stream": "^2.3.7",
     "webpack": "^4.44.2",
     "webpack-cli": "^3.3.10"
+  },
+  "resolutions": {
+    "stream-browserify": "^3.0.0"
```
Comment (jiexi) [inline package.json:63]: needed for `stream.pipeline` since webpack v4 uses v2 still. Another reason for us to get up to date with webpack

**Comment by jiexi** [comment]
Screenshot 2023-09-29 at 10 54 43 AM

**Comment by jiexi** [comment]
`@metamask/providers`
v8: we are here
v9: doesn't affect us
v10: I don't see METAMASK_EXTENSION_STREAM_CONNECT used anywhere anymore in extension. All i see is [this](https://github.com/MetaMask/metamask-extension/blob/8fcd83bb0462e9c5b13159f3a9a24c2c25e16341/app/scripts/contentscript.js#L577C39-L577C39) in regards to retry. [Searching the whole org yields nothing much](https://github.com/search?q=org%3AMetaMask%20METAMASK_EXTENSION_STREAM_CONNECT&type=code)

File: src/inpage/webpack.config.js (hunk 1 above)
Comment (jiexi) [reply src/inpage/webpack.config.js:16]: 🙃

**Comment by adonesky1** [comment]
https://github.com/MetaMask/mobile-provider/pull/85#issuecomment-1741286883

Just to be clear, this is meant to indicate that when pulled into a local mobile build the changes are indeed present and working, correct @jiexi ?

**Comment by jiexi** [comment]
@adonesky1 Yes, exactly that.

**Comment by jiexi** [comment]
@SocketSecurity ignore extension-port-stream@2.1.1
@SocketSecurity ignore @metamask/safe-event-emitter@3.0.0

**Comment by jiexi** [comment]
@SocketSecurity ignore unicode-canonical-property-names-ecmascript@2.0.0
@SocketSecurity ignore unicode-match-property-ecmascript@2.0.0

Google's npm bot

**Comment by jiexi** [comment]
@SocketSecurity ignore update-browserslist-db@1.0.13

wrappers around specific commands to npx

**Comment by jiexi** [comment]
@SocketSecurity ignore @babel/plugin-syntax-import-meta@7.10.4

unmaintained

**Review by sethkfman — APPROVED** [review]
LGTM

**Review by rekmarks — APPROVED** [review]

**Pull request merged**